src/library/clementine/clementinefeature.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <QFuture>
+#include <QFutureWatcher>
+#include <QStringListModel>
+#include <QtConcurrentRun>
+#include <QtSql>
+
+#include "library/baseexternallibraryfeature.h"
+#include "library/clementine/clementinedbconnection.h"
+#include "library/trackcollectionmanager.h"
+#include "library/treeitem.h"
+#include "library/treeitemmodel.h"
+
+class ClementinePlaylistModel;
+
+class ClementineFeature : public BaseExternalLibraryFeature {
+    Q_OBJECT
+  public:
+    ClementineFeature(Library* pLibrary, UserSettingsPointer pConfig);
+    virtual ~ClementineFeature();
+    static bool isSupported();
+    static void prepareDbPath(UserSettingsPointer pConfig);
+
+    virtual QVariant title();
+    virtual QIcon getIcon();
+
+    virtual TreeItemModel* getChildModel();
+
+  public slots:
+    virtual void activate();
+    virtual void activateChild(const QModelIndex& index);
+
+  private:
+    virtual void appendTrackIdsFromRightClickIndex(QList<TrackId>* trackIds, QString* pPlaylist);
+
+    ClementinePlaylistModel* m_pClementinePlaylistModel;
+    TreeItemModel m_childModel;
+    QStringList m_playlists;
+
+    //a new DB connection for the worker thread
+    ClementineDbConnection m_connection;
+
+    QSqlDatabase m_database;
+    bool m_isActivated;
+    QString m_dbfile;
+
+    QFutureWatcher<TreeItem*> m_future_watcher;
+    QFuture<TreeItem*> m_future;
+    QString m_title;
+    bool m_cancelImport;
+    QIcon m_icon;
+
+    static QString m_databaseFile;
```

Comment:
Mutable static members are a bad idea.

Reply:
added this in https://github.com/mixxxdj/mixxx/pull/3173/commits/f2a59de119ecefa97cf8d9ae8821a79d37650a05